crates/sui-core/src/authority_server.rs

```diff
@@ -588,66 +579,35 @@ impl ValidatorService {
         let tx_digest = transaction.digest();
         let _span = error_span!("validator_state_submit_transaction", ?tx_digest);
 
-        let tx_output = state
+        // TODO(fastpath): Skip checking for execution effects
+        // Effects are useless in mfp if they cannot be mapped to a consensus
+        // position
+        let _ = state
```

Comment:
So I think that if we already have executed effects (which means it has been executed through a finalized path), we should be able to return the results as well, instead of resubmitting it to get a new position. In fact, most likely `handle_vote_transaction` will fail because we won't be able to sign the same transaction again as they are no longer available.

Reply:
OK based on our discussion, this seems reasonable. We no longer nee `handle_vote_transaction` to return anything though. We could just check whether it has executed there.